File: Demos/C#/CustomOpenSaveDialogs/SaveDialogForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace CustomOpenSaveDialogs
{
  public partial class SaveDialogForm : Form
  {
    public string ReportName
    {
      get
      {
        return tbReportName.Text;
      }
    }
    
    public SaveDialogForm()
    {
      InitializeComponent();
    }

    private void tbReportName_TextChanged(object sender, EventArgs e)
    {
      btnOK.Enabled = !String.IsNullOrEmpty(ReportName);
    }
  }
}

File: Demos/VB.Net/CustomOpenSaveDialogs/SaveDialogForm.vb
Public Class SaveDialogForm
    ' Properties
    Public ReadOnly Property ReportName() As String
        Get
            Return tbReportName.Text
        End Get
    End Property

    Private Sub tbReportName_TextChanged(ByVal sender As Object, ByVal e As EventArgs) Handles tbReportName.TextChanged
        btnOK.Enabled = Not String.IsNullOrEmpty(Me.ReportName)
    End Sub
End Class
